front50-core/src/test/groovy/com/netflix/spinnaker/front50/model/pipeline/PipelineSpec.groovy

```diff
@@ -0,0 +1,81 @@
+package com.netflix.spinnaker.front50.model.pipeline
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.front50.api.model.pipeline.Pipeline
+import com.netflix.spinnaker.front50.api.model.pipeline.Trigger
+import com.netflix.spinnaker.front50.jackson.mixins.PipelineMixins
+import spock.lang.Specification
+
+class PipelineSpec extends Specification {
+  ObjectMapper objectMapper = new ObjectMapper()
+
+  void setup() {
+    objectMapper.addMixIn(Pipeline.class, PipelineMixins.class)
+  }
+
+  def 'should ignore correct pipeline properties from deserializing Pipeline object to JSON'() {
+    given:
+    objectMapper.addMixIn(Pipeline.class, PipelineMixins.class)
+    Pipeline pipeline = new Pipeline()
+    pipeline.setId("1")
+    String pipelineJSON = objectMapper.writeValueAsString(pipeline)
+
+    expect:
+    pipelineJSON == '{"lastModifiedBy":null,"triggers":[],"lastModified":null,"createdAt":null}'
+  }
+
+  def 'should ignore correct pipeline properties from serializing JSON to Pipeline object'() {
+    given:
+    objectMapper.addMixIn(Pipeline.class, PipelineMixins.class)
+    String pipelineJSON = '{"foo": "bar"}'
+
+    Pipeline pipeline = objectMapper.readValue(pipelineJSON, Pipeline.class)
+
+    expect:
+    pipeline.getType() == null
+  }
+
+  def 'should set any additional pipeline properties when serializing JSON to Pipeline object'() {
+    given:
+    objectMapper.addMixIn(Pipeline.class, PipelineMixins.class)
+    String pipelineJSON = '{"foo": "bar"}'
+
+    Pipeline pipeline = objectMapper.readValue(pipelineJSON, Pipeline.class)
+
+    expect:
+    pipeline.getAny() == [foo: "bar"]
+  }
+
+  def 'should set additional pipeline properties when deserializing JSON to Pipeline object'() {
```

Comment:
I would call this one out, since it's important:

`'roundtrip (JSON -> Pipeline -> JSON) retains arbitrary values'`